### PR TITLE
[VLMs] add helpers for get/set embedding

### PR DESCRIPTION
**Pull request by zucchini-nlp**
# What does this PR do?

Fixes (partially) https://github.com/huggingface/transformers/issues/38121, and verifies all VLMs have their respective getters/setters for embeddings and the decoder.

### CODE REVIEW

**Review by Cyrilvallez — APPROVED** [review]
LGTM in principle, however I know e.g. `generate` tries to use the `get_decoder` if it can to prepare masks in advance if it finds a `_prepare_4d_causal_attention_mask_with_cache_position` in it, so it may have some side effects (there was a few models that had these methods at the wrong place, leading to issues recently I think) -> may be nice to double check
